### PR TITLE
`tbl_uvregression()` to accept survey objects

### LINKED ISSUE #742: Add support for survey objects with `tbl_uvregression()`

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**
* `tbl_uvregression()` now accepts both data frames and survey objects as input.
closes #742

```r
tt <- survey::svydesign(~1, data = as.data.frame(Titanic), weights = ~Freq) 

tbl <- 
  tt %>%
  tbl_uvregression(
    y = Survived,
    method = survey::svyglm,
    method.args = list(family = binomial),
    hide_n = TRUE
  )
tbl
```
image

--------------------------------------------------------------------------------

Checklist for PR reviewer

- [x] PR branch has pulled the most recent updates from master branch. Ensure the pull request branch and your local version match and both have the latest updates from the master branch.
- [x] If an update was made to `tbl_summary()`, was the same change implemented for `tbl_svysummary()`?
- [x] If a new function was added, function included in `_pkgdown.yml`
- [x] If a bug was fixed, a unit test was added for the bug check
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Code coverage is suitable for any new functions/features. Review coverage with `covr::report()`. Before you run, set `Sys.setenv(NOT_CRAN="true")` and begin in a fresh R session without any packages loaded. 
- [x] R CMD Check runs without errors, warnings, and notes
- [x] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged into master:
- [x] Update NEWS.md with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parantheses at the end update (see `NEWS.md` for examples).
- [x] Increment the version number using `usethis::use_version(which = "dev")` 
- [x] Run `codemetar::write_codemeta()`
- [x] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

**Comment by ddsjoberg** [comment]
@larmarange would you mind reviewing this pull request please? Anytime we're working with survey objects, I feel somewhat nervous! Thank you!

File: R/tbl_uvregression.R [hunk 1]
```diff
@@ -25,6 +25,7 @@
 #'
 #' @param data Data frame to be used in univariate regression modeling.  Data
 #' frame includes the outcome variable(s) and the independent variables.
+#' Survey design is also accepted.
 #' @param method Regression method (e.g. [lm], [glm], [survival::coxph], and more).
```
Comment (larmarange) [inline R/tbl_uvregression.R:29]: add [survey::svyglm] in the list of possibilities ?

File: R/tbl_uvregression.R [hunk 2]
```diff
@@ -106,6 +107,12 @@ tbl_uvregression <- function(data, method, y = NULL, x = NULL, method.args = NUL
     )
```
Comment (larmarange) [inline R/tbl_uvregression.R:107]: Add a third example in the examples using a survey object

**Review by larmarange — APPROVED** [review]
For me it looks OK. However, as is, it appears to be a hidden feature. I would simply update documentation. (cf comments)

Otherwise, this is great!!! :-)

**Comment by ddsjoberg** [comment]
@larmarange thank you so so so much!

**Pull request merged**